app/views/application/_navigation_edit_my_application.html.haml

```diff
@@ -1,2 +1,6 @@
-%li.menu-item
-  = link_to 'Min ansökan', edit_membership_application_path(membership_app)
+- if current_user.is_member?
```

Comment:
it's fine to use a normal `if..then..else` if you want.  Important thing is to DRY it up.